lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -910,6 +913,39 @@ export class UserAgentApplication {
     return adalFrame;
   }
 
+  private pollWindowForHash(contentWindow: Window): Promise<string> {
+      return new Promise((resolve, reject) => {
+          const intervalTimeMs = 50;
+          const maxTicks = this.config.system.loadFrameTimeout / intervalTimeMs;
+          let ticks = 0;
+
+          const intervalId = setInterval(() => {
+              try {
+                  // Will throw if cross origin
+                  const href = contentWindow.location.href;
+
+                  // Don't process blank pages
+                  if (href === "" || href === "about:blank") {
+                      return;
+                  }
+
+                  // Only start clock when we are on same domain
+                  ticks++;
+
+                  if (contentWindow.closed) {
+                      clearInterval(intervalId);
+                 } else if (this.urlContainsHash(href)) {
+                      clearInterval(intervalId);
+                      resolve(contentWindow.location.hash);
+                  } else if (ticks > maxTicks) {
+                      clearInterval(intervalId);
+                      reject(ClientAuthError.createTokenRenewalTimeoutError()); // better error?
+                  }
+              } catch (e) {}
```

Comment:
what happens here?

Reply:
This is to catch the cross origin errors that will happen when we try to access the location object of the STS login UI pages, which we want to catch and ignore.

Reply:
move the `try{} catch {}` to smaller scope. Maybe only try and catch the `contentWindow.location.href`? And leave a comment explaining the silent `catch`

Reply:
I get what you're saying, I think I was trying to be safe here, as the only errors we want to happen here are timeout errors that we create.

Reply:
Agree about adding the comment, though.